### PR TITLE
executor: return error when region cache is invalid in fast analyze

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- return error when region cache is invalid in fast analyze

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -735,6 +735,9 @@ func (e *AnalyzeFastExec) buildSampTask() (needRebuild bool, err error) {
 
 	store, _ := e.ctx.GetStore().(tikv.Storage)
 	e.cache = store.GetRegionCache()
+	if e.cache == nil {
+		return false, errors.Trace(errors.New("region cache is invalid, fast analyze abort"))
+	}
```
Comment (eurekaka) [inline executor/analyze.go:740]: Shall we check the result of `e.cache.GetRPCContext(bo, loc.Region)` in `getSampRegionsRowCount` as well?

**Review by eurekaka — COMMENTED** [review]
LGTM

**Review by ichn-hu — APPROVED** [review]
LGTM

**Comment by zz-jason** [comment]
@lzmhhh123 could you add a test to ensure the bug is fixed and not happen again?

**Pull request closed without merging**